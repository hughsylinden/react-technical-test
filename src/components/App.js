import "../styles/App.css";
import React from "react";
import Search from "./Search";

function App() {
  return (
    <div className="App">
      <img
        className="App-image"
        alt="nasalogo"
        src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
      />
      <Search />
    </div>
  );
}

export default App;
