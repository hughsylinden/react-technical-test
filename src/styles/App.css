.App{
  display:flex;
  flex-direction: column;
}

.App-image{
  margin: auto;
  width: 500px;
  height: 300px;
}