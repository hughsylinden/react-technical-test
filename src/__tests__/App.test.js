import { render } from '@testing-library/react';
import App from '../components/App';

describe('App', () => {

  it('renders correctly', ()=> {
    const { getByAltText,asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
    expect(getByAltText('nasalogo')).toHaveClass("App-image");
  })
})
