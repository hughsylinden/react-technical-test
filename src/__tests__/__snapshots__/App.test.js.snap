// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <img
      alt="nasalogo"
      class="App-image"
      src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    />
    <form
      class="search-form"
    >
      <input
        class="search-input"
        data-testid="search-bar"
        type="text"
      />
      <button
        class="search-button"
        name="search-name"
        type="submit"
      >
        search
      </button>
    </form>
  </div>
</DocumentFragment>
`;
